pydatastructs/linear_data_structures/arrays.py

```diff
@@ -133,6 +134,117 @@ def __len__(self):
     def __str__(self):
         return str(self._data)
 
+class MultiDimensionalArray(Array):
+    """
+    Represents a multi-dimensional array.
+
+    Parameters
+    ==========
+
+    dtype: type
+        A valid object type.
+    *args: int
+        The dimensions of the array.
+
+    Raises
+    ======
+
+    IndexError
+        Index goes out of boundaries, or
+        the number of index given is not
+        the same as the number of dimensions.
+    ValueError
+        When there's no dimensions or the
+        dimension size is 0.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import MultiDimensionalArray as MDA
+    >>> arr = MDA(int, 5, 6, 9)
+    >>> arr.fill(32)
+    >>> arr[3, 0, 0]
+    32
+    >>> arr[3, 0, 0] = 7
+    >>> arr[3, 0, 0]
+    7
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Array_data_structure#Multidimensional_arrays
+
+    """
+    __slots__ = ['_sizes', '_data', '_dtype']
+
+    def __new__(cls, dtype=NoneType, *args, **kwargs):
+        if dtype is NoneType or len(args) == (0):
+            raise ValueError("Array cannot be created due to incorrect"
+                             " information.")
+        if len(args) == 1:
+            obj = Array.__new__(cls)
+            obj._dtype = dtype
+            obj._shape = [1]
+            obj._data = [None] * args[0]
+            return obj
+        dimensions = args
+        for dimension in dimensions:
+            if dimension < 1:
+                raise ValueError("Array cannot be created due to incorrect"
+                                 " dimensions.")
+        n_dimensions = len(dimensions)
+        d_sizes = []
+        index = 0
+        while n_dimensions > 1:
+            size = dimensions[index]
+            for i in range(index,  len(dimensions) - 1):
+                size = size * dimension
+            d_sizes.append(size)
+            n_dimensions -= 1
+            index += 1
+        d_sizes.append(dimensions[index])
+        d_sizes.append(1)
+        obj = Array.__new__(cls)
+        obj._dtype = dtype
+        obj._shape = d_sizes
+        obj._data = [None] * obj._shape[1] * dimensions[0]
+        return obj
+
```

Comment:
```suggestion
    @classmethod
    def methods(cls):
        return ['__new__', '__getitem__', '__setitem__', 'fill']

```

Reply:
Please append `pyds.MultiDimensionalArray` to the following list,
https://github.com/codezonediitj/pydatastructs/blob/9d063dd61a37438b49df218628d58b093c5239d2/pydatastructs/utils/tests/test_code_quality.py#L99